cdc/sink/validator.go

```diff
@@ -96,3 +106,63 @@ func preCheckSinkURI(sinkURIStr string) error {
 
 	return nil
 }
+
+func checkBDRMode(ctx context.Context, sinkURI string, replicaConfig *config.ReplicaConfig) error {
+	maskSinkUri, err := util.MaskSinkURI(sinkURI)
+	if err != nil {
+		return err
+	}
+	isDownstreamTiDBOrMySQL, err := checkIsDownstreamTiDBoRMySQL(sinkURI)
+	if err != nil {
+		return err
+	}
+	if !isDownstreamTiDBOrMySQL {
+		return cerror.ErrSinkURIInvalid.
+			GenWithStack("sink uri scheme is not supported in BDR mode, sink uri: %s", maskSinkUri)
+	}
+	cfg := pmysql.NewConfig()
+	id := model.DefaultChangeFeedID("sink-verify")
+	uri, err := url.Parse(sinkURI)
+	if err != nil {
+		return cerror.WrapError(cerror.ErrSinkURIInvalid, err)
+	}
+	err = cfg.Apply(ctx, id, uri, replicaConfig)
+	if err != nil {
+		return err
+	}
+	dsn, err := pmysql.GenBasicDSN(uri, cfg)
+	if err != nil {
+		return err
+	}
+	testDB, err := pmysql.GetTestDB(ctx, dsn, pmysql.CreateMySQLDBConn)
+	if err != nil {
+		return err
+	}
+	defer testDB.Close()
+	supported, err := pmysql.CheckIfSupportBDRMode(ctx, testDB)
+	if err != nil {
+		return err
+	}
+	if !supported {
+		// TODO: refine this error message
+		return cerror.ErrSinkURIInvalid.
+			GenWithStack("downstream database does not support BDR mode, "+
+				"please check your config, sink uri: %s", maskSinkUri)
+	}
+	return nil
+}
+
+func checkIsDownstreamTiDBoRMySQL(sinkURI string) (bool, error) {
```

Comment:
We already have the same functionality in `IsMySQLCompatibleScheme` in `pkg/sink/sink.go`.

Reply:
thanks